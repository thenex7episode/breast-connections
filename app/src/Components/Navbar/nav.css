nav {
    /* border-bottom: 1px solid black; */
    background-color: #F433AB; 
    color: white;
    position: fixed;
    z-index: 5;
    width:100%;
    display: flex;
    justify-content: space-between;
}

.deskNav {
    display: flex;
    margin: 0;
    padding: 1em;
}

nav > ul > li {
    padding: 1em;
}

nav > ul > li > a{
    color: white;
}

.navIcon {
    padding: 0 0.3em;
    color: white;
}

.hamburger {
    display: none;
    padding: 1.2em;
    color: white;
    font-size: 2.2em;
    cursor: pointer;
}

.mobNav {
    background-color: #F433AB; 
    position: fixed;
    padding: 8em 0;
    height: 100vh;
    z-index: 4;
    transition: 1s;
}

.showMob {
    right: 0;
}

.mobNav > li {
    padding: 0.5em 2em;
}

@media (max-width: 900px){
    .deskNav {
        display: none;
    }
    .hamburger {
        display: block;
    }
    .mobNav {
        display: block
    }
}