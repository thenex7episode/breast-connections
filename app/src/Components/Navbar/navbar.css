
 

.logo{
    float: left;
    height:100px;
    width:100px;
    margin-left: 25px;
}
ul{
    cursor: pointer;
    font-size: 1.5em;
}
nav{
    padding-right: 10em;
}
ul li{
    float: right;
    width: 200px;
    height: 40px;
    color: white;
    opacity: .8;
    line-height: 40px;
    text-align: center;
}
.forums{
    position: fixed;
    top: 5em;
    background-color: purple;
    right: 2em;
}
.sub-menu{
    display: block;
    font-size: .4em;
}
.sub-menu:hover{
background:#FF88DC;
}
.sub-wrapper{
    display:none;
}
.sub-menu:hover .sub-wrapper{
    display: block;
}
.logo{
    float: left;
    height:4.5em;
    width:4.5em;
    margin-left: 65px;
    margin-top: 5px;
}
.nav {
    border-bottom: 1px solid black;
    text-align: right;
    line-height: 70px;
    background-color:#F433AB; 
    color: white;
    min-height:5em;
}
.link {
    margin: 0 30px 0 0;
}
.menu{
    display:flex;
    flex-direction: row;
    justify-content: flex-end;
}
.menu a {
    
    clear: right;
    text-decoration: none;
    margin: 0 10px;
    line-height: 70px;
    color: white;
    font-size: 1.3em;
    font-family: Arial, Helvetica, sans-serif;
}

label {
    margin: 0 40px 0 0;
    font-size: 26px;
    line-height: 70px;
    display: none;
    width: 26px;
    float: right;
}
#toggle {
    display: none;
}



@media only screen and (max-width: 768px) {
    .logo{
        height:3em;
        width:3em;
        margin-top: 1em;

    }
    label {
        display: block;
        cursor: pointer;
    }
    .menu {
        text-align: center;
        width: 100%;
        display: none;
    }
    .menu a {
        display: block;
        border-bottom: 1px solid #EAEAEB;
        margin: 0;
        color: black;
     
    }
    #toggle:checked + .menu {
        display: block;
    }
    
    
    
}





