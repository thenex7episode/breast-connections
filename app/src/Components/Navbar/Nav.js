import React, { Component } from 'react';
import axios from 'axios';
import './nav.css'
import { Link } from 'react-router-dom';
import textLogo from '../textlogo1.png';
import { Menu, Dropdown, Icon, Button, Avatar } from 'antd';
const SubMenu = Menu.SubMenu;


const menu = (
    <Menu>
      <Menu.Item>
        <Link to='/dashboard/motivation'>Motivation</Link>
      </Menu.Item>
      <Menu.Item>
        <Link to='/dashboard/family'>Family</Link>
      </Menu.Item>
      <Menu.Item>
        <Link to='/dashboard/resources'>Resources</Link>
      </Menu.Item>
      <Menu.Item>
        <Link to='/dashboard/nutrition'>Nutrition</Link>
      </Menu.Item>
      <Menu.Item>
        <Link to='/dashboard/health'>Health</Link>
      </Menu.Item>
      <Menu.Item>
        <Link to='/dashboard/financial'>Financial</Link>
      </Menu.Item>
    </Menu>
  );


export default class Nav extends Component {
    constructor(){
        super();
        this.state = {
            username: '',
            user_id: null,
            nav: false
        }
    }

    logout= () => {
        axios.post('/logout').then(() => {
        window.location = '/'
        }).catch(e => {console.log('Logout error', e)})
      }

    componentDidMount(){
        axios.get('/api/check-session').then(r => {
              if(r.data.username) {
                  this.setState({username: r.data.username, user_id: r.data.user_id, image: r.data.imageurl})
              } else {
                  console.log('not logged in')
                  this.props.history.location.push('/login')
              }
          }).catch(err => {console.log('WWEEEWWWOOOWWWEEEWWWOOO',err)})
      }

    render() {
        return (
            <div>
            <nav>
                <div><img src={textLogo} alt="logo" style={{width: '250px'}}/></div>
                <ul className='deskNav'>
                    <li><Icon className='navIcon' type="laptop" /><Link to={this.state.username ? '/dashboard' : '/'}>Home</Link></li>
                    <Dropdown overlay={menu}>
                    <li><Icon className='navIcon' type="team" /><Link to='/forums'>Forums</Link></li>
                    </Dropdown>
                    <li><Icon className='navIcon' type="environment" /><Link to='/search'>Search</Link></li>
                    <li><Icon className='navIcon' type="shopping-cart" /><Link to='/shop'>Shop</Link></li>
                    <li><Icon className='navIcon' type="bulb" /><Link to='/info'>Info</Link></li>
                    <li style={{display: this.state.user_id ? 'none' : 'block'}}><Link to='/login'><Button icon="user">Login</Button></Link></li>
                    <Dropdown overlay={<Menu>
                                            <Menu.Item>
                                                <Icon type='user' />
                                                <Link to={`/profile/${this.state.username}`}>Profile</Link>
                                            </Menu.Item>
                                            <Menu.Item>
                                                <Icon type='chat' />
                                                <Link to={`/chatroom/`}>Chat</Link>
                                            </Menu.Item>
                                            <Menu.Item>
                                                <Icon type='logout' />
                                                <Link onClick={() => this.logout()}to='/dashboard/motivation'>Logout</Link>
                                            </Menu.Item>
                                        </Menu>}>
                    <Link to={`/profile/${this.state.username}`}><Avatar style={{display: this.state.user_id ? 'block' : 'none', margin: '0.7em'}} size="large" src={this.state.image} /></Link>
                    </Dropdown>
                </ul>
                <Icon className='hamburger' onClick={() => this.setState({nav: !this.state.nav})} type={this.state.nav ? 'menu-unfold' : 'menu-fold'} />
            </nav>
            <ul style={{right: this.state.nav ? '0' : '-100%'}} className='mobNav'>
                <li><Icon className='navIcon' type="laptop" /><Link to={this.state.username ? '/dashboard' : '/'}>Home</Link></li>
                <li><Icon className='navIcon' type="team" /><Link to='/dashboard'>Forums</Link></li>
                <li><Icon className='navIcon' type="environment" /><Link to='/search'>Search</Link></li>
                <li><Icon className='navIcon' type="shopping-cart" /><Link to='/shop'>Shop</Link></li>
                <li><Icon className='navIcon' type="bulb" /><Link to='/info'>Info</Link></li>
                <li><Icon className='navIcon' type="chat" /><Link to='/chatroom'>Chat</Link></li>
                <li style={{display: this.state.user_id ? 'none' : 'block'}}><Link to='/login'><Button icon="user">Login</Button></Link></li>
            </ul>
            </div>
        );
    }
}