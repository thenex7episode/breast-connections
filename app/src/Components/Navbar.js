import React, { Component } from 'react';
import './navbar.css'
import {Link } from 'react-router-dom'
import logo from '../logo.png'
import axios from'axios'


export default class Navbar extends Component {
  constructor() {
    super() 

    this.state = {
      isLoggedIn: false
    }
    console.log('isLoggedIn', this.state.isLoggedIn)
  }



  componentWillReceiveProps() {
    axios.get('/api/check-session').then(r => {
      console.log('hello from the mount side')
        if(r.data.username) {
            console.log('navbar username log', r.data.username)
            this.setState({isLoggedIn: true})
        }
    }).catch(err => {console.log('WWEEEWWWOOOWWWEEEWWWOOO',err)})
}


logout= () => {
  axios.post('/logout').then(() => {
    this.setState({isLoggedIn: false})
  }).catch(e => {console.log('Logout error', e)})
}

render() {
  const {isLoggedIn} = this.state
  console.log('isLggediniogho;',isLoggedIn)
  return (
      
    <div>
            <div className='nav'>
                <img src={logo} alt="logo" className="logo" />
                    <label for='toggle'>&#9776;</label>
                        <input type='checkbox' id='toggle' />
                            <div className='menu'>
                                <a> {!isLoggedIn ? <Link className="link" to="/login">Login</Link> :
                                    <Link onClick={this.logout}className="link" to="/">Logout</Link>}
                                </a>

                                {/* DROPDOWN MENU */}
                                <div className='dropdown'>
                                <a> <Link className="link" to="/forums"> Forums
                                  </Link>
                                  <div class="dropdown-content">
                                    <a href="#">Motivation</a>
                                    <a href="#">Family</a>
                                    <a href="#">Resources</a>
                                  </div>
                                </a>
                                </div>

                                {/* END DROPDOWN MENU */}

                                <a> <Link className="link" to="/info"> Info
                            </Link>
                            </a>
                        <a><Link className="link" to="/home">
                        Home
                    </Link></a>
            </div>
        </div>
    </div>

        )}
        }

        