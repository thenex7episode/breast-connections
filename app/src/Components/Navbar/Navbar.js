import React, { Component } from 'react';
import './navbar.css'
import {Link, Redirect } from 'react-router-dom'
import logo from '../logo.png'
import axios from'axios';
import { Menu, Dropdown, Icon } from 'antd';
import { browserHistory } from 'react-router'



export default class Navbar extends Component {
  constructor() {
    super() 

    this.state = {
      isLoggedIn: false
    }
    console.log('isLoggedIn', this.state.isLoggedIn)
  }


  // manually redirecting with react router dom - this doesnt seem to work
  componentDidMount(){
    axios.get('/api/check-session').then(r => {
          if(r.data.username) {
              this.setState({isLoggedIn: true})
          } else {
              console.log('not logged in')
              this.props.history.push('/login')
          }
      }).catch(err => {console.log('WWEEEWWWOOOWWWEEEWWWOOO',err)})
  }


//   componentWillReceiveProps() {
//     axios.get('/api/check-session').then(r => {
//       console.log('hello from the mount side')
//         if(r.data.username) {
//             console.log('navbar username log', r.data.username)
//             this.setState({isLoggedIn: true})
//         }
//     }).catch(err => {console.log('WWEEEWWWOOOWWWEEEWWWOOO',err)})
// }


logout= () => {
  axios.post('/logout').then(() => {
    this.setState({isLoggedIn: false})
   window.location.reload(true)
  }).catch(e => {console.log('Logout error', e)})
}

render() {
  const {isLoggedIn} = this.state
  console.log('isLggediniogho;',isLoggedIn)
  return (
      
    <div>
            <div className='nav'>
                <img src={logo} alt="logo" className="logo" />
                            <div className='menu'>
                                <a> {!isLoggedIn ? <Link className="link" to="/login">Login</Link> :
                                    <Link onClick={this.logout}className="link" to=''>Logout</Link>}
                                </a>
                                <a> <Link className="link" to="/forums"> Forums
                                  </Link>
                                </a>
                                <a> <Link className="link" to="/info"> Info
                            </Link>
                            </a>
                        <a><Link className="link" to="/">
                        Home
                    </Link></a>
            </div>
        </div>
    </div>

        )}
        }

        