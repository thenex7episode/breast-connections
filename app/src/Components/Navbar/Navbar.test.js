import Navbar from './Navbar'

it("renders correctly", () => {
    const wrapper = shallow(
        <Navbar />
    );
    expect(wrapper).toMatchSnapshot()
});

