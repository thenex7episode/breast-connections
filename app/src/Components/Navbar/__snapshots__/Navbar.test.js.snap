// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="nav"
  >
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
    <img
      alt="logo"
      className="textlogo"
      src="textlogo.png"
    />
    <label
      for="toggle"
    >
      â˜°
    </label>
    <input
      id="toggle"
      type="checkbox"
    />
    <div
      className="menu"
    >
      <a>
         
        <Link
          className="link"
          replace={false}
          to="/login"
        >
          Login
        </Link>
      </a>
      <div
        className="dropdown"
      >
        <a>
           
          <Link
            className="link"
            replace={false}
            to="/forums"
          >
             Forums
          </Link>
          <div
            class="dropdown-content"
          >
            <a
              href="/dashboard/family"
            >
              Family
            </a>
            <a
              href="/dashboard/resources"
            >
              Resources
            </a>
            <a
              href="/dashboard/motivation"
            >
              Motivation
            </a>
            <a
              href="/dashboard/nutrition"
            >
              Nutrition
            </a>
            <a
              href="/dashboard/health"
            >
              Health
            </a>
            <a
              href="/dashboard/financial"
            >
              Financial
            </a>
          </div>
        </a>
      </div>
      <a>
         
        <Link
          className="link"
          replace={false}
          to="/info"
        >
           Info
        </Link>
      </a>
      <a>
        <Link
          className="link"
          replace={false}
          to="/"
        >
          Home
        </Link>
      </a>
    </div>
  </div>
</div>
`;
